### PR TITLE
fix: non-primitive parameter types are converted to query parameters

### PR DESCRIPTION
**Pull request by Agreon**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #1197 

## What is the new behavior?

non-primitive path parameters are exported as type string.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by TBG-FR** [comment]
Encountered that issue today, I almost ripped my hair off on that one ahah, hope it'll be merged soon !

**Comment by Agreon** [comment]
Seems like the v8 release is more important at the moment :wink:

File: lib/services/schema-object-factory.ts [hunk 1]
```diff
@@ -110,6 +110,15 @@ export class SchemaObjectFactory {
...
     }
+
+    // Path parameters are always of type string.
+    if (param.in === 'path') {
+      return {
+        ...param,
+        type: 'string'
+      };
+    }
+
```
Comment (kamilmysliwiec) [inline lib/services/schema-object-factory.ts:121]: What if I have a class that represents all path parameters and I use it as follows: `@Path() params: EndpointPathParams`. I want its properties to _be spread_ and represent multiple path parameters. Would this PR still support this case?

Comment (Agreon) [reply lib/services/schema-object-factory.ts:121]: Sorry for the late response, just had time to investigate: I couldn't find a decorator named `@Path` in the nestjs project. 
Did you mean `@Param`?. If that's the case we have a conflict here. The questions is, who is using this decorator that way.

Comment (kamilmysliwiec) [reply lib/services/schema-object-factory.ts:121]: I meant `@Param()`. As for who is using this decorator that way, I suppose pretty much everyone (given that this is how it's described in the docs)

**Comment by Agreon** [comment]
Closing this PR as my assumptions about the plugin were wrong.

**Comment by MaorBarazani2** [comment]
> Closing this PR as my assumptions about the plugin were wrong.

Is there any update on this one? expected in a future version, or tossed aside for now?

**Comment by Clashsoft** [comment]
This is still a problem, and apparently the tracking point for this kind of issue (#1197, #2471).

**Comment by Clashsoft** [comment]
I found some janky solution, using `pnpm patch` to edit the `SchemaObjectFactory`:

```patch
--- a/dist/services/schema-object-factory.js
+++ b/dist/services/schema-object-factory.js
@@ -53,6 +53,9 @@ class SchemaObjectFactory {
         if ((0, is_date_ctor_util_1.isDateCtor)(param.type)) {
             return Object.assign(Object.assign({ format: 'date-time' }, param), { type: 'string' });
         }
+        if (param.type.name === 'ObjectId') {
+            return Object.assign(Object.assign({ format: 'objectid' }, param), { type: 'string' });
+        }
         if ((0, lodash_1.isFunction)(param.type)) {
             const propertiesWithType = this.extractPropertiesFromType(param.type, schemas);
             if (!propertiesWithType) {
```

**Pull request closed without merging**